blueflood-elasticsearch/src/test/java/com/rackspacecloud/blueflood/io/ElasticIOIntegrationTest.java

```diff
@@ -312,39 +395,103 @@ public void testDeDupMetrics() throws Exception {
         elasticIO.setINDEX_NAME_WRITE(ElasticIOConfig.ELASTICSEARCH_INDEX_NAME_WRITE.getDefaultValue());
     }
 
-    private MetricNameSearchIO getDiscoveryIO(String type) {
-        if (type.equalsIgnoreCase("elasticTokensIO")) {
-            return elasticTokensIO;
-        } else {
-            return elasticIO;
+    @Test
+    public void testRegexLevel0() {
+        List<String> terms = Arrays.asList("foo", "bar", "baz", "foo.bar", "foo.bar.baz", "foo.bar.baz.aux");
+
+        List<String> matchingTerms = new ArrayList<>();
+        Pattern patternToGet2Levels = Pattern.compile(elasticIO.regexToGrabCurrentAndNextLevel("*"));
+        for (String term: terms) {
+            Matcher matcher = patternToGet2Levels.matcher(term);
+            if (matcher.matches()) {
+                matchingTerms.add(term);
+            }
+        }
+
+        assertEquals(1, matchingTerms.size());
+        assertEquals("foo.bar", matchingTerms.get(0));
+    }
+
+    @Test
+    public void testRegexLevel1() {
+        List<String> terms = Arrays.asList("foo", "bar", "baz", "foo.bar", "foo.bar.baz", "foo.bar.baz.aux");
+
+        List<String> matchingTerms = new ArrayList<String>();
+        Pattern patternToGet2Levels = Pattern.compile(elasticIO.regexToGrabCurrentAndNextLevel("foo.*"));
+        for (String term: terms) {
+            Matcher matcher = patternToGet2Levels.matcher(term);
+            if (matcher.matches()) {
+                matchingTerms.add(term);
+            }
         }
+
+        assertEquals(2, matchingTerms.size());
+        assertEquals("foo.bar", matchingTerms.get(0));
+        assertEquals("foo.bar.baz", matchingTerms.get(1));
     }
 
     @Test
-    @Parameters({"elasticIO", "elasticTokensIO"})
```

Comment:
I noticed that because of this, it's missing coverage on one method. I will add test cases to cover getMetricNames() which is in AbstractElasticIO class.